orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -0,0 +1,67 @@
+package org.reactome.orthoinference;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import static org.gk.model.ReactomeJavaConstants.*;
+
+/*
+ *  All PhysicalEntitys, ReactionlikeEvents and Pathways are routed to this class to generate their stable identifiers
+ */
+public class StableIdentifierGenerator {
+
+    private static MySQLAdaptor dba;
+    private static GKInstance instanceEditInst;
+    private static Map<String,Integer> seenOrthoIds = new HashMap<>();
+    private static String speciesAbbreviation = null;
+
+    public static GKInstance generateOrthologousStableId(GKInstance inferredInst, GKInstance originalInst) throws Exception {
+
+        // All Human PhysicalEntitys and Events will have a StableIdentifier instance in the stableIdentifier attribute
+        GKInstance stableIdentifierInst = (GKInstance) originalInst.getAttributeValue(stableIdentifier);
```

Comment:
It might be good to have a check here to see if `stableIdentifierInst` is null and log a fatal error with an exception being thrown if so.

Reply:
I'll add that in. Should it be fatal to orthoinference though? Is it because if a human instance doesn't have a stable identifier, then that is a bigger problem that needs to be addressed?

Reply:
Yeah, it probably should be fatal to orthoinference.  Potentially, we could flag it and continue with other inferences, but it might indicate a bigger underlying problem that would be harder to patch after all the rest of the inference has happened.

We should probably investigate immediately to see why that situation happened.

Reply:
I vote for :skull: ...with a check that runs at the before orthoinference to ensure that it should enver actually happen.

Reply:
I think that's probably best.